src/modules/endpoint/DefaultDataContainer.js

```diff
@@ -31,133 +59,84 @@ class DefaultDataContainer implements DataContainer {
     return this._cities
   }
 
-  getCategoriesMap = async (): Promise<CategoriesMapModel> => {
-    if (this._context === null) {
-      throw Error('Context has not been set yet.')
-    }
-    if (this._categoriesMap === null) {
-      throw Error('CategoriesMap is null.')
-    }
-    return this._categoriesMap
+  getCategoriesMap = (context: DatabaseContext): Promise<CategoriesMapModel> => {
+    const cache: Cache<Array<EventModel>> = this.caches['categories']
+    return cache.get(context)
   }
 
-  getEvents = async (): Promise<Array<EventModel>> => {
-    if (this._context === null) {
-      throw Error('Context has not been set yet.')
-    }
-    if (this._events === null) {
-      throw Error('Events are null.')
-    }
-    return this._events
+  getEvents = (context: DatabaseContext): Promise<Array<EventModel>> => {
+    const cache: Cache<Array<EventModel>> = this.caches['events']
+    return cache.get(context)
   }
 
-  getLanguages = async (): Promise<Array<LanguageModel>> => {
-    if (this._context === null) {
-      throw Error('Context has not been set yet.')
-    }
-    if (this._languages === null) {
-      throw Error('Languages are null.')
-    }
-    return this._languages
+  getLanguages = async (context: DatabaseContext): Promise<Array<LanguageModel>> => {
```

Comment:
From here on you use `async`, in the above occurences you don't.

Reply:
I removed async where I didn't need it.